### ISSUE TITLE
Unit id

### ISSUE THREAD

**Issue by FLO-2DKaren** [FLO-2DSoftware/FLO-2DMapCrafter#34]
Units Let's consider how to add Unit identifier to the maps or legends. Like ft or m for depth or what ever units specific energy comes in.

**Comment by rpachaly** [comment]
This gave me an idea. What do you think about adding a tab called "Summary" where we add all the project relevant variables? Like:

Units: 
Build n°: 
Grid Size: 
N° of elements:
Simulation type:
Simulation date:

We could also use this tab to add some settings to MapCrafter like ID labels for scenarios or anything like that.

**Comment by FLO-2DKaren** [comment]
Yes a great idea for sure.  Can you write that data to a geopackage table called summary?  That way you don't lose it when you close MapCrafter or the project..  That data could also be added to the map description table for one of the mapping layouts.  The one with the large description box.

**Comment by rpachaly** [comment]
> Can you write that data to a geopackage table called summary? That way you don't lose it when you close MapCrafter or the project.

I'm not an expert of using geopackage (yet). But I will figure this out and let you know. 

> That data could also be added to the map description table for one of the mapping layouts. The one with the large description box.

Very good idea. I'll code this.

**Comment by FLO-2DKaren** [comment]
Cool I think packaging the vector and table data in a geopackage would be faster and less data than shapefiles.  Plus you only have one file for a lot of data.

**Comment by rpachaly** [comment]
> Cool I think packaging the vector and table data in a geopackage would be faster and less data than shapefiles. Plus you only have one file for a lot of data.

I agree. But how the geopackage deal with rasters? This is my concern.

**Comment by FLO-2DKaren** [comment]
I think it only vector and table data can be used in geopackage.  The rest would be raster data.

The only other way to speed things up would probably be to use lazy load and the mesh tools.  The mesh tools act like rasters.  But the mesh tools are kind of limited in that I'm not sure how you publish them to a map layout with visible layers because you can't see all of the groups in the mesh.

**Comment by rpachaly** [comment]
We can use rasters on the geopackage. Check this out [MapCrafter.zip](https://github.com/FLO-2DSoftware/FLO-2DMapCrafter/files/13241199/MapCrafter.zip). But I need to figure out a way to keep track of the groups and symbology.

I think it is better to focus on the other issues/bugs right now and add the geopackage in a future version.

**Comment by FLO-2DKaren** [comment]
Oh that's cool!!!  I agree to wait on it.  I'm looking into style-db manager.  I think we can add a project with our own styles to the style database.  I just haven't figured out how to do it yet.

**Comment by rpachaly** [comment]
> Oh that's cool!!! I agree to wait on it. I'm looking into style-db manager. I think we can add a project with our own styles to the style database. I just haven't figured out how to do it yet.

That`s a good idea! I've never worked with the style database, but it shouldn`t be difficult to figure out.

**Comment by FLO-2DKaren** [comment]
You can save the style to the gpkg tables and default that style to a specific layer.  Check this video out at 8:30 time frame
https://youtu.be/2N5ZyFlbqSQ?feature=shared&t=510

**Comment by FLO-2DKaren** [comment]
Units:
Build n°:
Grid Size:
N° of elements:
Simulation type:
Simulation date:
Coordinate system EPSG code

**Comment by FLO-2DKaren** [comment]
Ignore most of the comments in this issue.

**Comment by CorneliusKibet** [comment]
@FLO-2DKaren, 

Do we need to show all the line, e.g, "Pro Model - Build No. 23.10.25", or just extract "23.10.25" as in the case below?

<img width="922" height="689" alt="Image" src="https://github.com/user-attachments/assets/d1f24c42-f13f-4753-8a6e-99bfa0a24efb" />

**Comment by FLO-2DKaren** [comment]
Just the number is ok.

**Comment by CorneliusKibet** [comment]
@FLO-2DKaren, 

Hello,

Please find the attached .gif file showing the progress on this issue.

Regards.

![Image](https://github.com/user-attachments/assets/87b51965-34c8-4add-b26a-6afbd731e902)

**Comment by CorneliusKibet** [comment]
@FLO-2DKaren, 

Actually there's one missing thing, Project ID. How do we go about it?

**Comment by FLO-2DKaren** [comment]
I generally use Project ID when I load maps but from a different run.  So it makes a different mapcrafter folder in that run directory. 
So it still saves the text file to the mapcrafter folder of the correct project but the maps that are loaded on QGIS have a project id so the end user can tell the difference between two projects.

So I think you can update your summary anytime the Path is changed or updated. 

But I don't think you need to add a Project ID to your summary.

**Comment by CorneliusKibet** [comment]
Thanks for the clarification.
